datalad/distributed/ora_remote.py

```diff
@@ -167,9 +170,64 @@ def write_file(self, file_path, content, mode='w'):
 
         raise NotImplementedError
 
+    def make_readonly(self, path):
+        """Revoke write permissions for `path`
+
+        Parameters
+        ----------
+        path: Path or list of Path
+        """
+
+        # NOTE on VFAT: This will silently fail to remove write permission
+        # for the owner of a directory on VFAT, which implies that files
+        # still can be deleted w/o a "permission denied". The silence on that
+        # failure to have an effect is shared between `chmod` (via SSH) and
+        # `pathlib.Path.chmod` (local IO).
+        raise NotImplementedError
+
 
 class LocalIO(IOBase):
     """IO operation if the object tree is local (e.g. NFS-mounted)"""
+
+    @staticmethod
+    @contextmanager
+    def ensure_write_permission_parent(destination):
```

Comment:
I think such a helper should live outside this class and outside `ora_remote.py`

Reply:
This will require substantial RF'ing (it needs the `IO` classes to do shit). That should happen, yes, but not in this PR. This wi need to be a dedicated one (likely not just one).

Reply:
How does it need the `IO` class? It makes no single reference to it. And being here makes it harder to test, and less visible. Is there any advantage of having it in `LocalIO`?

Reply:
Well, yes - in case of the `LocalIO` it doesn't directly depend on it (as in doesn't use `self`). But it has a twin in the SSH class, which needs to use its `self`. Higher-level functions are not supposed to distinguish which IO they are using. Hence, nominally this function needs to stay. So - I could factor out the local case in the sense that the core of it becomes a helper outside and then this method here is just a wrapper around it. And the SSH/shell variant stays where it is. I thought you meant both. If you only want to have this one separate - can do.

Reply:
In that case, I'd go for a general "request write permission for `path`" and leave the "parent thereof" notion in here. Agree?

Reply:
@mih 

Pushed Proposal: https://github.com/datalad/datalad/pull/6493/commits/60151778b59005c9039d8c37b84fd65f7e63da6c